apps/docs-snippets/src/guide/predicates/interacting-with-predicates.test.ts

```diff
@@ -0,0 +1,149 @@
+      gasLimit: 2000,
+      maxFee: bn(0),
+    });
+
+    const predicateCoins = await predicate.getResourcesToSpend([
+      { amount: 2000, assetId: baseAssetId },
+    ]);
+
+    // Add the predicate input and resources
+    transactionRequest.addResources(predicateCoins);
+    // #endregion interacting-with-predicates-1
+
+    expect(transactionRequest.inputs.length).toBeGreaterThanOrEqual(1);
+    expect(transactionRequest.outputs.length).toEqual(1);
+  });
+
+  it('should successfully transfer funds to the predicate', async () => {
+    const transactionRequest = new ScriptTransactionRequest({ gasLimit: 2000, maxFee: bn(0) });
+    transactionRequest.addCoinOutput(receiver.address, 100, baseAssetId);
+
+    const txCost = await provider.getTransactionCost(transactionRequest, {
+      resourcesOwner: predicate,
+    });
+
+    transactionRequest.gasLimit = txCost.gasUsed;
+    transactionRequest.maxFee = txCost.maxFee;
+
+    await predicate.fund(transactionRequest, txCost);
+
+    // #region interacting-with-predicates-2
+
+    const result = await predicate.sendTransaction(transactionRequest);
+
+    await result.waitForResult();
+    // #endregion interacting-with-predicates-2
+
+    const { isStatusSuccess } = await result.waitForResult();
+
+    expect(isStatusSuccess).toBeTruthy();
+  });
+
+  it('should successfully simulate a transaction with predicate', async () => {
+    // #region interacting-with-predicates-3
+    const transactionRequest = new ScriptTransactionRequest({ gasLimit: 2000, maxFee: bn(0) });
+    transactionRequest.addCoinOutput(receiver.address, 100, baseAssetId);
+
+    const txCost = await provider.getTransactionCost(transactionRequest, {
+      resourcesOwner: predicate,
+    });
+
+    transactionRequest.gasLimit = txCost.gasUsed;
+    transactionRequest.maxFee = txCost.maxFee;
+
+    await predicate.fund(transactionRequest, txCost);
+
+    const result = await predicate.simulateTransaction(transactionRequest);
+
+    // #endregion interacting-with-predicates-3
+
+    expect(result.receipts).toEqual([
+      {
+        type: 1,
+        id: expect.any(String),
+        val: expect.any(BN),
+        pc: expect.any(BN),
+        is: expect.any(BN),
+      },
+      {
+        type: 9,
+        gasUsed: expect.any(BN),
+        result: expect.any(BN),
+      },
+    ]);
+  });
+
+  it('should successfully populate the transaction with predicate data', async () => {
+    const quantity: CoinQuantityLike[] = [[500, baseAssetId]];
+
+    // #region interacting-with-predicates-4
+    let transactionRequest = new ScriptTransactionRequest({ gasLimit: 2000, maxFee: bn(0) });
+    transactionRequest.addCoinOutput(receiver.address, 100, baseAssetId);
+
+    const predicateResources = await provider.getResourcesToSpend(predicate.address, quantity);
+
+    transactionRequest.addResources(predicateResources);
+
+    transactionRequest = predicate.populateTransactionPredicateData(transactionRequest);
```

Comment:
That's a good point,  I had added this before. Should we deprecate this method? I can't think of a use case for it otherwise. cc @Torres-ssf

Reply:
@maschad This method cannot be deprecated as it's still used by the `EVMWalletConnector`. Given the context of how this connector works, the predicate that it uses is only instantiated after its `sendTransaction` method is called with the TX already funded with predicate resources that were fetched as a normal account. So I suppose this connector will always use this method.

We also [use this method](https://github.com/FuelLabs/fuels-ts/blob/977d58ba50c55de64621c8d509e660937cb99991/packages/account/src/providers/provider.ts#L1141-L1143) within the TS SDK at `Provider.getTransactionCost`, to populate the fake resources we create for a dry-run, as they may belong to a Predicate. The issue https://github.com/FuelLabs/fuels-ts/issues/2362 addresses this problem

In your example, the Predicate instance is being used to fetch its resources, so we do not need to call `populateTransactionPredicateData`

Reply:
Thanks for the clarification, I suppose it may not make sense to expose this method in our public API given most consumers of the SDK would not have much use for it then.